br/pkg/stream/table_mapping.go

```diff
@@ -253,42 +389,140 @@ func (tm *TableMappingManager) parseTableIdAndUpdateIdMapping(
 	return nil
 }
 
-func (tm *TableMappingManager) parseTableValueAndUpdateIdMapping(dbID int64, value []byte,
-	collector MetaInfoCollector) error {
+func extractTableSimpleInfo(value []byte) (int64, *tableSimpleInfo, error) {
 	var tableInfo model.TableInfo
 	if err := json.Unmarshal(value, &tableInfo); err != nil {
+		return 0, nil, errors.Trace(err)
+	}
+	var partitionIds []int64
+	partitions := tableInfo.GetPartitionInfo()
+	if partitions != nil {
+		partitionIds = make([]int64, 0, len(partitions.Definitions))
+		for _, def := range partitions.Definitions {
+			partitionIds = append(partitionIds, def.ID)
+		}
+	}
+	return tableInfo.ID, &tableSimpleInfo{
+		Name:         tableInfo.Name.O,
+		PartitionIds: partitionIds,
+	}, nil
+}
+
+func (tm *TableMappingManager) parseTableValueAndUpdateIdMappingForDefaultCf(
+	dbID int64, value []byte, ts uint64) error {
+	tableId, tableSimpleInfo, err := extractTableSimpleInfo(value)
+	if err != nil {
 		return errors.Trace(err)
 	}
+	key := tableMetaKey{
+		dbId:    dbID,
+		tableId: tableId,
+		ts:      ts,
+	}
+	if existingValue, exists := tm.tempDefaultKVTableMap[key]; exists {
+		existingValue.count++
+		return nil
+	}
 
-	dbReplace, err := tm.getOrCreateDBReplace(dbID)
+	tm.tempDefaultKVTableMap[key] = &tableMetaValue{
+		info:  tableSimpleInfo,
+		count: 1,
+	}
+	return nil
+}
+
+func (tm *TableMappingManager) parseTableValueAndUpdateIdMappingForWriteCf(
+	dbId, tableId int64, value []byte, commitTs uint64, collector MetaInfoCollector) error {
+	rawWriteCFValue := new(RawWriteCFValue)
+	if err := rawWriteCFValue.ParseFrom(value); err != nil {
+		return errors.Trace(err)
+	}
+
+	// handle different write types
+	if rawWriteCFValue.IsDelete() || rawWriteCFValue.IsRollback() {
+		// for delete operations, we should clear the tempTableInfo if it exists
+		// but not process it as a table creation/update
+		idx := tableMetaKey{
+			dbId:    dbId,
+			tableId: tableId,
+			ts:      rawWriteCFValue.GetStartTs(),
+		}
+		delete(tm.tempDefaultKVTableMap, idx)
```

Comment:
if count > 0 here is it still need to be deleted?

Reply:
since using start ts as key, it's okay here. 

but is it easy to cause OOM, we may need more tests here.